src/type-checker.cc

```diff
@@ -436,12 +433,23 @@ Result TypeChecker::OnBrTableTarget(Index depth) {
   // signatures.
   if (br_table_sig_ == nullptr) {
     br_table_sig_ = &label_sig;
-  }
-  if (*br_table_sig_ != label_sig) {
-    result |= Result::Error;
-    PrintError("br_table labels have inconsistent types: expected %s, got %s",
-               TypesToString(*br_table_sig_).c_str(),
-               TypesToString(label_sig).c_str());
+  } else {
+    if (features_.reference_types_enabled()) {
+      if (br_table_sig_->size() != label_sig.size()) {
+        result |= Result::Error;
+        PrintError("br_table labels have inconsistent arity: expected %" PRIzd
+                   " got %" PRIzd,
+                   br_table_sig_->size(), label_sig.size());
```

Comment:
I'm confused about where we check that the types are compatible beyond their arity. In the algorithm.rst doc, this is done by a call to `pop_vals`, but where is it done here?

Reply:
That already happens above in the call to `CheckSignature` .. that basically ensure that the value stack contains the values that match the label.  That part doesn't change.